docs/ai/tutorials/snippets/evaluate-with-reporting/MyTests.cs

```diff
@@ -0,0 +1,153 @@
+    {
+        const string SystemPrompt =
+            """
+        You're an AI assistant that can answer questions related to astronomy.
+        Keep your responses concise and try to stay under 100 words.
+        Use the imperial measurement system for all measurements in your response.
+        """;
+
+        IList<ChatMessage> messages =
+            [
+                new ChatMessage(ChatRole.System, SystemPrompt),
+            new ChatMessage(ChatRole.User, astronomyQuestion)
+            ];
+
+        var chatOptions =
+            new ChatOptions
+            {
+                Temperature = 0.0f,
+                ResponseFormat = ChatResponseFormat.Text
+            };
+
+        ChatResponse completion = await chatClient.GetResponseAsync(messages, chatOptions);
+        return (messages, completion);
+    }
+    // </SnippetGetResponse>
+
+    // <SnippetValidate>
+    /// <summary>
+    /// Runs basic validation on the supplied <see cref="EvaluationResult"/>.
+    /// </summary>
+    private static void Validate(EvaluationResult result)
+    {
+        // Retrieve the score for relevance from the <see cref="EvaluationResult"/>.
+        NumericMetric relevance =
+            result.Get<NumericMetric>(RelevanceTruthAndCompletenessEvaluator.RelevanceMetricName);
+        Assert.IsFalse(relevance.Interpretation!.Failed);
+        Assert.IsTrue(relevance.Interpretation.Rating is EvaluationRating.Good or EvaluationRating.Exceptional);
+
+        // Retrieve the score for truth from the <see cref="EvaluationResult"/>.
+        NumericMetric truth = result.Get<NumericMetric>(RelevanceTruthAndCompletenessEvaluator.TruthMetricName);
+        Assert.IsFalse(truth.Interpretation!.Failed);
+        Assert.IsTrue(truth.Interpretation.Rating is EvaluationRating.Good or EvaluationRating.Exceptional);
+
+        // Retrieve the score for completeness from the <see cref="EvaluationResult"/>.
+        NumericMetric completeness =
+            result.Get<NumericMetric>(RelevanceTruthAndCompletenessEvaluator.CompletenessMetricName);
+        Assert.IsFalse(completeness.Interpretation!.Failed);
+        Assert.IsTrue(completeness.Interpretation.Rating is EvaluationRating.Good or EvaluationRating.Exceptional);
+
+        // Retrieve the word count from the <see cref="EvaluationResult"/>.
+        NumericMetric wordCount = result.Get<NumericMetric>(WordCountEvaluator.WordCountMetricName);
+        Assert.IsFalse(wordCount.Interpretation!.Failed);
+        Assert.IsTrue(wordCount.Interpretation.Rating is EvaluationRating.Good or EvaluationRating.Exceptional);
+        Assert.IsFalse(wordCount.ContainsDiagnostics());
+        Assert.IsTrue(wordCount.Value <= 100);
+    }
+    // </SnippetValidate>
+
+    // <SnippetTestMethod>
+    [TestMethod]
+    public async Task SampleAndEvaluateResponse()
+    {
+        // Create a <see cref="ScenarioRun"/> with the scenario name
+        // set to the fully qualified name of the current test method.
```

Comment:
Please feel free to reword - but it would be nice to call out the `await using` statement since forgetting to include it can cause confusion (since the result will not be persisted if this is missed).

```suggestion
        // set to the fully qualified name of the current test method.
        // The use of await using below ensures that the <see cref="ScenarioRun"/>
        // will be correctly disposed at the end and that the results of this evaluation
        // will be correctly persisted to the result store.
```